File: iteration2/Models/Person.java
package iteration2.Models;

public abstract class Person extends Model {
    private String name;
    private String surname;
    private String ssn;
    private Character gender;

    public Person(String name, String surname, String ssn, Character gender) {
        this.name = name;
        this.surname = surname;
        this.ssn = ssn;
        this.gender = gender;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSurname() {
        return surname;
    }

    public void setSurname(String surname) {
        this.surname = surname;
    }

    public String getSsn() {
        return ssn;
    }

    public void setSsn(String ssn) {
        this.ssn = ssn;
    }

    public Character getGender() {
        return gender;
    }

    public void setGender(Character gender) {
        this.gender = gender;
    }
}

File: iteration3/Models/Person.py
import json
from datetime import date
from abc import ABC, abstractmethod

from iteration3.Models.Model import Model


class Person(Model, ABC):
    def __init__(self, name: str, surname: str, ssn: str, gender: str):
        super().__init__()
        self.__name = name
        self.__surname = surname
        self.__ssn = ssn
        self.__gender = gender

    @property
    def name(self):
        return self.__name

    @property
    def surname(self):
        return self.__surname

    @property
    def ssn(self) -> str:
        return self.__ssn

    @ssn.setter
    def ssn(self, ssn: str):
        self.__ssn = ssn

    @property
    def gender(self) -> str:
        return self.__gender

    @gender.setter
    def gender(self, gender: str):
        self.__gender = gender
